stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewDescriptor.java

```diff
@@ -51,4 +95,35 @@ private static String getIdAttribute(View element) {
   public View getViewForHighlighting(Object element) {
     return (View)element;
   }
+
+  private class ElementContext {
+    private View mElement;
+
+    public void hook(View element) {
+      mElement = Util.throwIfNull(element);
+    }
+
+    public void unhook() {
+      if (mElement != null) {
+        mElement = null;
+      }
+    }
+
+    private void setAttributeAsText(Map<String, String> attributeToValueMap) {
+      for (Map.Entry<String, String> entry : attributeToValueMap.entrySet()) {
+        String methodName = "set" + capitalize(entry.getKey());
+        String propertyValue = entry.getValue();
+        mMethodInvoker.invoke(mElement, methodName, propertyValue);
+      }
+    }
+
+    private String capitalize(String str) {
+      if (str == null || str.length() == 0) {
+        return str;
+      }
+      char[] buffer = str.toCharArray();
```

Comment:
I believe it's better to round-trip this through a `StringBuilder` instead of `char[]`, e.g.
```
StringBuilder buffer = new StringBuilder(str);
buffer.setCharAt(0, Character.toTitleCase(buffer.charAt(0));
return buffer.toString();
```
It seems equivalent, but `StringBuilder` has an important optimization that saves an extra copying of the string. It creates a `String` that aliases the same internal `char[]`, and then also switches to copy-on-write mode. So it's saving an extra copy and a temporary allocation.